### DISCUSSION TITLE
[Camera Support]: Recording is buggy after using Live View in Frigate 0.14

### DISCUSSION THREAD

**Discussion by katschub88** [blakeblackshear/frigate#12850] [Camera Support] [answered]
### Describe the problem you are having

I've just updated Frigate to 0.14 and at first everything seemed to work fine.
But then I noticed that the latest recordings won't play.

When Frigate is started, everything works fine. But when I open the Live View of my camera (which also plays fine), errors are created in system log (see below).
Usually the recorded mp4 files are 10 seconds each, but after opening the Live View (and the errors appear), the recorded mp4 files are 1 second each and there is no audio track, only video. There doesn't seem to be any video footage missing, it's just split into 1 second chunks. Just the audio is missing in every 1 second chunk.

Also: In order to get to the recording view, I have to open the live view first, which causes the bugs.
Now when I restart frigate (via console) while still in the recording view, I can view the new recordings without any problems.
It's just the live view for some reason that triggers the problem.
I can see the errors in the log and the mp4 files which have only 10% of the usual size.

Before the upgrade to 0.14, every prior version of Frigate was working without any problems.

### Steps to reproduce

See description...

### Version

0.14.0-da913d8

### Frigate config file

```yaml
go2rtc:
  modes:
    - webrtc
  streams:
    cam:
      - rtsp://user:password@ip:port/h264Preview_01_main
      - ffmpeg:rtsp://user:password@ip:port/h264Preview_01_main#audio=opus

cameras:
  cam:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/cam?video&audio
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/cam?video=copy
          input_args: preset-rtsp-restream
          roles:
            - detect
      output_args:
        record: preset-record-generic-audio-copy

...
```

### Relevant log output

```shell
error	frigate.video	cam: Unable to read frames from ffmpeg process.
error	frigate.video	cam: ffmpeg process is not running. exiting capture thread...
error	watchdog.cam	Ffmpeg process crashed unexpectedly for cam.
error	watchdog.cam	The following ffmpeg logs include the last 100 lines prior to exit.
error	ffmpeg.cam.detect	[rtsp @ 0x64afe5bfbc40] RTP: PT=60: bad cseq 5829 expected=ac21
error	ffmpeg.cam.detect	[out_0_0 @ 0x64afe5d8aac0] 100 buffers queued in out_0_0, something may be wrong.
error	ffmpeg.cam.detect	[out_0_0 @ 0x64afe5d8aac0] 1000 buffers queued in out_0_0, something may be wrong.
warning	frigate.record.maintainer	Unable to keep up with recording segments in cache for cam. Keeping the 6 most recent segments out of 7 and discarding the rest...
warning	frigate.record.maintainer	Unable to keep up with recording segments in cache for cam. Keeping the 6 most recent segments out of 11 and discarding the rest...
```

### Operating system

Debian

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

Reolink Doorbell WiFi

**Comment by hawkeye217** [comment, marked as answer]
You should be using the recommended Reolink config for your doorbell: https://docs.frigate.video/configuration/camera_specific/#reolink-doorbell

**Comment by katschub88** [comment]
I've tested it with the flv stream and the problem is the same.
Also it's been working perfectly fine so far with prior versions and the latest doorbell firmware.

**Reply by hawkeye217** [reply]
Many other users have the Reolink doorbell configured and working with 0.14. If you set up yours using the method as outlined in the docs, what error messages do you get in your frigate logs?

**Reply by NickM-27** [reply]
Nothing has changed with ffmpeg. A 1 second recording will be created in the case that the camera stream sends incorrect timestamps so ffmpeg is insane to cut the segments at the correct time. 

I'd suggest starting with the recommended config then we can look at logs

**Reply by katschub88** [reply]
Ok, so now after a system update and full reboot it seems to work with the flv config.

What I don't understand is: Why was it working in every prior version of Frigate and is now causing a problem with live view? Which stream is live view using? Was is using the restream (127.0.0.1:8554) before and is now using the camera stream directly?

I remember that I have switched from flv to rtsp a while ago, after installing the latest firmware of the doorbell. The flv stream was causing problems (no signal or something and i think there was also another issue with 2-way audio) while the rtsp stream has been working perfectly fine for many months now.

**Reply by katschub88** [reply]
And one more point: If I don't open live view, everything is working without any problems. Recording is fine, detection is fine, etc.
So there still is a problem with live view that hasn't been there before and possibly goes undetected with the flv stream.
It would be interesting to see if the used streams have changed from 0.13 live view to 0.14 live view.
Or maybe a change with the restream?
If I'm not mistaken, Frigate should use just one stream to the camera and then use the local restream for everything else. Or not?

**Reply by NickM-27** [reply]
The problem with your config above is that you used the full camera URL for the audio transcode, so this would actually make two connections to the camera. The recommended config only makes one connection to the camera.

**Reply by katschub88** [reply]
I thought I was using the flv stream at the moment, but now I've seen in the config that this is not the case.
I was experimenting with the stream config, trying to figure out what was causing the problem and somehow ended up with the flv stream commented out and the following config:

```
- ffmpeg:rtsp://user:password@ip:port/h264Preview_01_main #audio=opus
- rtsp://user:password@ip:port/h264Preview_01_main
```

It's almost the same as the original one, except that the "#audio=opus" part is commented out and the ffmpeg one is in the first place. Everything works fine so far, no problems with live view, nothing.
I totally thought it was the flv stream running without issues now. :sweat_smile:

I do remember that I've switched from flv stream to rtsp, because I encountered severe problems with the flv one. It's been a while, so I don't really remember what it was, but I was worried that I would run into the same issues again now.
I guess, I'll leave it like that for now, as it's working fine.

Thanks for the help, case closed.
